p2p/client_test.go

```diff
@@ -63,34 +87,204 @@ func TestBootstrapping(t *testing.T) {
 	require.NotEmpty(cid2)
 
 	// client1 has no seeds
-	client1, err := NewClient(config.P2PConfig{ListenAddress: "/ip4/127.0.0.1/tcp/7676"}, privKey1, logger)
+	client1, err := NewClient(context.Background(), config.P2PConfig{ListenAddress: "/ip4/0.0.0.0/tcp/7676"}, privKey1, logger)
 	require.NoError(err)
 	require.NotNil(client1)
 
 	// client2 will use client1 as predefined seed
-	client2, err := NewClient(config.P2PConfig{
+	client2, err := NewClient(context.Background(), config.P2PConfig{
 		ListenAddress: "/ip4/127.0.0.1/tcp/7677",
-		Seeds:         "/ip4/127.0.0.1/tcp/7676/p2p/" + cid1.Pretty(),
+		Seeds:         "/ip4/127.0.0.1/tcp/7676/p2p/" + cid1.String(),
 	}, privKey2, logger)
 	require.NoError(err)
 
 	// client3 will use clien1 and client2 as seeds
-	client3, err := NewClient(config.P2PConfig{
+	client3, err := NewClient(context.Background(), config.P2PConfig{
+		ListenAddress: "/ip4/127.0.0.1/tcp/7678",
+		Seeds:         "/ip4/127.0.0.1/tcp/7676/p2p/" + cid1.String() + ",/ip4/127.0.0.1/tcp/7677/p2p/" + cid2.String(),
+	}, privKey3, logger)
+	require.NoError(err)
+
+	// client4 will use clien1 as seed
+	client4, err := NewClient(context.Background(), config.P2PConfig{
+		ListenAddress: "/ip4/127.0.0.1/tcp/7679",
+		Seeds:         "/ip4/127.0.0.1/tcp/7676/p2p/" + cid1.String(),
+	}, privKey4, logger)
+	require.NoError(err)
+
+	err = client1.Start()
+	defer client1.Close()
+	assert.NoError(err)
+
+	err = client2.Start()
+	defer client2.Close()
+	assert.NoError(err)
+
+	err = client3.Start()
+	defer client3.Close()
+	assert.NoError(err)
+
+	err = client4.Start()
+	defer client4.Close()
+	assert.NoError(err)
```

Comment:
What about moving Client creation logic to a separate func to avoid repetitive code for every Start, NewClient, Generate and etc

Reply:
I would love to have a function to create entire test networks with given "topology" (initial connections setup using seed-nodes). Created issue for this: https://github.com/lazyledger/optimint/issues/27